src/promptflow/promptflow/_sdk/_service/apis/flow.py

```diff
@@ -1,68 +1,170 @@
 # ---------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
+import base64
+import os
+import uuid
+import json
+from flask import make_response, request, send_from_directory
+
 from promptflow._sdk._configuration import Configuration
-from promptflow._sdk._service import Namespace, Resource
+from promptflow._sdk._service import Namespace, Resource, fields
 from promptflow._sdk._service.utils.utils import get_client_from_request
+from promptflow._sdk._constants import UX_INPUTS_JSON
+from promptflow._sdk._utils import json_load, read_write_by_user, json_dump
 from promptflow._utils.yaml_utils import load_yaml
 from promptflow._utils.flow_utils import resolve_flow_path
 from pathlib import Path
 
 api = Namespace("Flows", description="Flows Management")
 
-# Define flow test request parsing
-flow_test_parser = api.parser()
-flow_test_parser.add_argument("flow", type=str, location="form", required=True)
-flow_test_parser.add_argument("node", type=str, location="form", required=False)
-flow_test_parser.add_argument("variant", type=str, location="form", required=False)
-flow_test_parser.add_argument("detail", type=str, location="form", required=False, default=None)
-flow_test_parser.add_argument("experiment", type=str, location="form", required=False)
-flow_test_parser.add_argument("inputs", type=dict, location="form", required=False, default={})
-flow_test_parser.add_argument("environment_variables", type=dict, location="form", required=False, default={})
-
-# Response model of flow operation
+
 dict_field = api.schema_model("FlowDict", {"additionalProperties": True, "type": "object"})
 
+flow_test_model = api.model(
+    "FlowTest",
+    {
+        "flow": fields.String(required=True, description="Path to flow directory to test."),
+        "node": fields.String(required=False, description="If specified it will only test this node, else it will "
+                                                          "test the flow."),
+        "variant": fields.String(required=False, description="Node & variant name in format of ${"
+                                                             "node_name.variant_name}, will use default variant if "
+                                                             "not specified."),
+        "detail": fields.String(required=False, description="Output path of flow detail"),
+        "experiment": fields.String(required=False, description="Path of experiment template"),
+        "inputs": fields.Nested(dict_field, required=False),
+        "environment_variables": fields.Nested(dict_field, required=False),
+    },
+)
+
+image_save_model = api.model(
+    "ImageSave",
```

Comment:
We might support other MIME types, like audio video. We could rename this as MediaSave?